app/src/main/java/com/example/poste/activities/FolderViewActivity.java

```diff
@@ -72,77 +50,85 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // Prep vars
         TextView emptyText = findViewById(R.id.folderViewEmptyText);
+        TextView folderName = findViewById(R.id.folderNameText);
+        ImageButton newBut = findViewById(R.id.newPost);
+        ImageButton settingsBut = findViewById(R.id.folderSettings);
+        Button openBut = findViewById(R.id.buttonOpen);
+        Button editBut = findViewById(R.id.buttonEdit);
+        Button deleteBut = findViewById(R.id.buttonDelete);
         postRecyclerView = findViewById(R.id.posts_recycler_view);
-        currentUser = PosteApplication.getCurrentUser();
-        registerForContextMenu(postRecyclerView);
 
-        // Get current folder
-        int currentFolderId = getIntent().getIntExtra("folderId", -1);
-        try {
-            currentFolder = API.getFolderById(currentFolderId);
-
-            // Remove empty text if posts exist in folder
-            if (currentFolder.getPosts().size() > 0) {
-                emptyText.setVisibility(View.GONE);
+        // Create listeners for the folder buttons
+        newBut.setOnClickListener(view -> {
+            Post newPost = new Post(PosteApplication.getCurrentUser().getId(), "", "", PosteApplication.getCurrentUser().getId());
```

Comment:
I think we should consider making this an API call, rather than creating a post locally. Otherwise, we put our client-side and backend out of sync. Perhaps a more abstracted solution would be to get the data necessary to send a post creation call to the backend, and then either updating all user data (as we do during dashboard / login) or updating the given folder

This could be accomplished via a method on the singleton User, createPost() or some such, that handles making the API call (see User.updateFoldersAndPosts() on dev branch)

Also, it looks like you may need to pull dev into your branch and reconcile changes

Reply:
Discussed below